### PR TITLE
Expose Exception.set_HResult as Public

### PR DESCRIPTION
**Pull request by luqunl**
This is to change Excetion.set_HResult from protected to public, so that others can set HResult value for a given exception.

Approved API: https://github.com/dotnet/corefx/issues/29696
Related CoreCLR PR(S.P.Corelib change): https://github.com/dotnet/coreclr/pull/18203

### CODE REVIEW

**Comment by danmosemsft** [comment]
Can you please include a simple test that sets it?

**Comment by luqunl** [comment]
> Can you please include a simple test that sets it?

Added.

> Also can you replace existing use like ex.SetErrorCode(__HResults.RO_E_CLOSED); ?

I am going to replace all of these in next PR.

**Comment by luqunl** [comment]
> Also can you replace existing use like ex.SetErrorCode(__HResults.RO_E_CLOSED); 

Replaced

File: src/System.Runtime.WindowsRuntime/src/System/InternalHelpers.cs [hunk 1]
```diff
@@ -1,16 +0,0 @@
...
-
-using System.Runtime.InteropServices;
-
-namespace System
-{
-    internal static class InternalHelpers
-    {
-        internal static void SetErrorCode(this Exception ex, int code)
-        {
-            InteropExtensions.SetExceptionErrorCode(ex, code);
```
Comment (danmosemsft) [inline src/System.Runtime.WindowsRuntime/src/System/InternalHelpers.cs]: This delete isn't sufficient to allow us to remove `[assembly: InternalsVisibleTo("System.Runtime.WindowsRuntime, PublicKey=00000000000000000400000000000000", AllInternalsVisible = false)]` from corelib I assume..

Comment (jkotas) [reply src/System.Runtime.WindowsRuntime/src/System/InternalHelpers.cs]: Right. It is getting us a bit closer though.

**Comment by luqunl** [comment]
@dotnet-bot test this

**Comment by luqunl** [comment]
@dotnet-bot test this please

**Comment by jkotas** [comment]
@luqunl This won't pass until you get both CoreCLR and CoreRT changes into CoreFX.

**Comment by jkotas** [comment]
Actually, CoreCLR is update is blocked by this. You will need to cherry pick this change into https://github.com/dotnet/corefx/pull/30001 so that this change goes together with the CoreCLR update. Let me know if you need help with this.

File: src/System.Runtime.WindowsRuntime/src/System.Runtime.WindowsRuntime.csproj [hunk 2]
```diff
@@ -78,13 +78,13 @@
     <Compile Include="System\Windows\Rect.cs" />
     <Compile Include="System\Windows\Size.cs" />
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetGroup)'=='uapaot'">
+  <ItemGroup Condition="'$(TargetGroup)'=='uapaot'">
```
Comment (jkotas) [inline src/System.Runtime.WindowsRuntime/src/System.Runtime.WindowsRuntime.csproj]: Nit: Unnecessary whitespace

**Comment by jkotas** [comment]
To make this build, you will need to ifdef `InternalHelpers.cs` temporarily as:

```
#if PROJECTN
    InteropExtensions.SetExceptionErrorCode(ex, code);
#else
    ex.HResult = code;
#endif
```

**Comment by luqunl** [comment]
> To make this build, you will need to ifdef InternalHelpers.cs temporarily as:

just add another file: InternalHelpers.CoreCLR.cs and InternalHelpers.CoreRT.cs

**Comment by luqunl** [comment]
@dotnet-bot test NETFX x86 release build

**Comment by luqunl** [comment]
XSLT failures are unrelated and already have an issue.
